modules/face/include/opencv2/face/face_alignment.hpp

```diff
@@ -0,0 +1,145 @@
+    may be used to endorse or promote products derived from this software
+    without specific prior written permission.
+
+This software is provided by the copyright holders and contributors "as is" and
+any express or implied warranties, including, but not limited to, the implied
+warranties of merchantability and fitness for a particular purpose are disclaimed.
+In no event shall copyright holders or contributors be liable for any direct,
+indirect, incidental, special, exemplary, or consequential damages
+(including, but not limited to, procurement of substitute goods or services;
+loss of use, data, or profits; or business interruption) however caused
+and on any theory of liability, whether in contract, strict liability,
+or tort (including negligence or otherwise) arising in any way out of
+the use of this software, even if advised of the possibility of such damage.*/
+#ifndef __OPENCV_FACE_ALIGNMENT_HPP__
+#define __OPENCV_FACE_ALIGNMENT_HPP__
+#include "opencv2/core.hpp"
+#include "opencv2/objdetect.hpp"
+#include "opencv2/imgproc.hpp"
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <fstream>
+#include <queue>
+#include <algorithm>
+#include <fstream>
+
+namespace cv{
+namespace face{
+class FacemarkKazemi
+{
+public:
+        /** @returns The number of landmarks detected in an image */
+        virtual int getNumLandmarks() const = 0;
+        /** @returns The number of faces detected in an image */
+        virtual int getNumFaces() const = 0;
+        /** @brief This function extract the data for training from .txt files which contain the corresponding image name and landmarks.
+        *Each file's first line should give the path of the image whose
+        *landmarks are being described in the file.Then in the subsequent
+        *lines there should be coordinates of the landmarks in the image
+        *i.e each line should be of the form "x,y"(ignoring the double quotes)
+        *where x represents the x coordinate of the landmark and y represents
+        *the y coordinate of the landmark.
+        *
+        *For reference you can see the files as provided in the
+        *<a href="http://www.ifp.illinois.edu/~vuongle2/helen/">HELEN dataset</a>
+        * @param filename A vector containing name of the .txt files.
+        * @param trainlandmarks A vector that would store shape or landmarks of all images.
+        * @param trainimages A vector which stores the images whose landmarks are tracked
+        * @returns A boolean value
+        */
+        virtual bool getData(std::vector<String> filename,std::vector< std::vector<Point2f> > & trainlandmarks
+                ,std::vector<String> & trainimages)=0;
+        /** @brief This function gets the relative shape of the face scaled and centered
+        according to the bounding rectangle of the detected face.
+        * @param face A vector storing rectangles of all  faces in the current image.
```

Comment:
watch out for double spaces, remove them as much as possible --> between all and faces.